integration-tests/smoke/ccip_messaging_test.go

```diff
@@ -165,18 +171,159 @@ func Test_CCIPMessaging(t *testing.T) {
 			[]byte("hello CCIPReceiver with low exec gas"),
-			ccipdeployment.MakeEVMExtraArgsV2(1, false), // 1 gas is too low.
-			ccipdeployment.EXECUTION_STATE_FAILURE,      // state would be failed onchain due to low gas
+			ccdeploy.MakeEVMExtraArgsV2(1, false), // 1 gas is too low.
+			ccdeploy.EXECUTION_STATE_FAILURE,      // state would be failed onchain due to low gas
 		)
+
+		manuallyExecute(ctx, t, state, destChain, out, sourceChain, e, sender)
+
+		t.Logf("successfully manually executed message %x",
+			out.msgSentEvent.Message.Header.MessageId)
 	})
 }
 
-func sleepAndReplay(t *testing.T, e ccipdeployment.DeployedEnv, sourceChain, destChain uint64) {
+func manuallyExecute(
+	ctx context.Context,
+	t *testing.T,
+	state ccdeploy.CCIPOnChainState,
+	destChain uint64,
+	out messagingTestCaseOutput,
+	sourceChain uint64,
+	e ccdeploy.DeployedEnv,
+	sender []byte,
+) {
+	merkleRoot := getMerkleRoot(
+		t,
+		state.Chains[destChain].OffRamp,
+		out.msgSentEvent.SequenceNumber,
+	)
+	messageHash := getMessageHash(
+		t,
+		state.Chains[destChain].OffRamp,
+		sourceChain,
+		out.msgSentEvent.SequenceNumber,
+		out.msgSentEvent.Message.Header.MessageId,
+	)
+	tree, err := merklemulti.NewTree(hashutil.NewKeccak(), [][32]byte{messageHash})
+	require.NoError(t, err)
+	proof, err := tree.Prove([]int{0})
+	require.NoError(t, err)
+	require.Equal(t, merkleRoot, tree.Root())
+
+	tx, err := state.Chains[destChain].OffRamp.ManuallyExecute(
+		e.Env.Chains[destChain].DeployerKey,
+		[]offramp.InternalExecutionReport{
+			{
+				SourceChainSelector: sourceChain,
+				Messages: []offramp.InternalAny2EVMRampMessage{
+					{
+						Header: offramp.InternalRampMessageHeader{
+							MessageId:           out.msgSentEvent.Message.Header.MessageId,
+							SourceChainSelector: sourceChain,
+							DestChainSelector:   destChain,
+							SequenceNumber:      out.msgSentEvent.SequenceNumber,
+							Nonce:               out.msgSentEvent.Message.Header.Nonce,
+						},
+						Sender:       sender,
+						Data:         []byte("hello CCIPReceiver with low exec gas"),
+						Receiver:     state.Chains[destChain].Receiver.Address(),
+						GasLimit:     big.NewInt(1),
+						TokenAmounts: []offramp.InternalAny2EVMTokenTransfer{},
+					},
+				},
+				OffchainTokenData: [][][]byte{
+					{},
+				},
+				Proofs:        proof.Hashes,
+				ProofFlagBits: boolsToBitFlags(proof.SourceFlags),
+			},
+		},
+		[][]offramp.OffRampGasLimitOverride{
+			{
+				{
+					ReceiverExecutionGasLimit: big.NewInt(200_000),
+					TokenGasOverrides:         nil,
+				},
+			},
+		},
+	)
+	_, err = deployment.ConfirmIfNoError(e.Env.Chains[destChain], tx, err)
+	require.NoError(t, err, "failed to send/confirm manuallyExecute tx")
+
+	receipt, err := e.Env.Chains[destChain].Client.TransactionReceipt(ctx, tx.Hash())
+	require.NoError(t, err)
+
+	var found bool
+	for _, lg := range receipt.Logs {
+		topic := offramp.OffRampExecutionStateChanged{}.Topic()
```

Comment:
This is good, but might be slightly less implementation dependent if you call get msg state function

Reply:
Oh that might be cleaner, good note.